apollo-rx2-support/src/main/java/com/apollographql/apollo/rx2/RxJavaExtensions.kt

```diff
@@ -0,0 +1,109 @@
+@file:Suppress("NOTHING_TO_INLINE")
+@file:JvmName("KotlinExtensions")
+
+package com.apollographql.apollo.rx2
+
+import com.apollographql.apollo.ApolloCall
+import com.apollographql.apollo.ApolloClient
+import com.apollographql.apollo.ApolloMutationCall
+import com.apollographql.apollo.ApolloPrefetch
+import com.apollographql.apollo.ApolloQueryCall
+import com.apollographql.apollo.ApolloQueryWatcher
+import com.apollographql.apollo.ApolloSubscriptionCall
+import com.apollographql.apollo.api.Mutation
+import com.apollographql.apollo.api.Operation
+import com.apollographql.apollo.api.Query
+import com.apollographql.apollo.api.Response
+import com.apollographql.apollo.api.Subscription
+import com.apollographql.apollo.cache.normalized.ApolloStoreOperation
+import io.reactivex.BackpressureStrategy
+import io.reactivex.Completable
+import io.reactivex.Flowable
+import io.reactivex.Observable
+import io.reactivex.Single
+import io.reactivex.annotations.CheckReturnValue
+
+@JvmSynthetic
+@CheckReturnValue
+inline fun ApolloPrefetch.rx(): Completable =
+    Rx2Apollo.from(this)
+
+@JvmSynthetic
+@CheckReturnValue
+inline fun <T> ApolloStoreOperation<T>.rx(): Single<T> =
+    Rx2Apollo.from(this)
+
+@JvmSynthetic
+@CheckReturnValue
+inline fun <T> ApolloQueryWatcher<T>.rx(): Observable<Response<T>> =
+    Rx2Apollo.from(this)
+
+@JvmSynthetic
+@CheckReturnValue
+inline fun <T> ApolloCall<T>.rx(): Observable<Response<T>> =
+    Rx2Apollo.from(this)
+
+@JvmSynthetic
+@CheckReturnValue
+inline fun <T> ApolloSubscriptionCall<T>.rx(
+    backpressureStrategy: BackpressureStrategy = BackpressureStrategy.LATEST
+): Flowable<Response<T>> = Rx2Apollo.from(this, backpressureStrategy)
+
+/**
+ * Creates a new [ApolloQueryCall] call and then converts it to an [Observable].
+ *
+ * The number of emissions this Observable will have is based on the
+ * [com.apollographql.apollo.fetcher.ResponseFetcher] used with the call.
+ */
+@JvmSynthetic
+@CheckReturnValue
+inline fun <D : Operation.Data, T, V : Operation.Variables> ApolloClient.rxQuery(
+    query: Query<D, T, V>,
+    configure: ApolloQueryCall<T>.() -> Unit = {}
+): Observable<Response<T>> = query(query).apply(configure).rx()
+
+/**
+ * Creates a new [ApolloMutationCall] call and then converts it to a [Single].
+ */
+@JvmSynthetic
+@CheckReturnValue
+inline fun <D : Operation.Data, T, V : Operation.Variables> ApolloClient.rxMutate(
```

Comment:
There should also be a `Single<T>` version for queries

Reply:
In our project, we have that. Should we add it for convenience? The problem is that there might be multiple emissions based on the configuration. If there are multiple emissions, then it will just crash.

Users can always use `toSingleOrError()` which makes it more explicit.

Reply:
I'm assuming there'd be multiple emissions if there was a cache involved? Are there any other uses cases that would trigger multiple emissions? I agree that explicit errors would be better, makes for less surprises

Reply:
My version of `toSingle` is a bit more involved.

```kotlin
data class ApolloErrorException(val errors: List<Error>) : Throwable()

val <T> Response<T>.exception: ApolloErrorException
  get() = ApolloErrorException(errors())

fun <T> Response<T>.responseToSingle(): Single<Response<T>> =
  if (hasErrors()) {
    Single.error(exception)
  } else {
    Single.just(this)
  }

fun <T> Observable<Response<T>>.toSingle(): Single<Response<T>> =
  retryWithBackoffAndJitter()
      .flatMapSingle { it.responseToSingle() }
      .singleOrError()
```

Reply:
Yes, multiple emissions would happen only when there is a cache.

I think for these extensions, it is more logical to keep them consistent with the original Java integration. That's why I don't want to have any custom logic in Kotlin extensions.

I think your suggestion is great. It would be great if you can contribute that back into Java implementation since it is not Kotlin-only.